CONTRIBUTING.md

```diff
@@ -1,81 +1,126 @@
-	   ```
-	   $ pack_model.sh --lm <language model> <tr_conf> <dec_conf> <cmvn> <e2e>
-	   ```
-	 - `pack_model.sh` also produces a packed espnet model (`model.tar.gz`). If you upload this model to somewhere with a download link,
-	   please put the link information to `RESULTS.md`.
-	 - please contact Shinji Watanabe <shinjiw@ieee.org> if you want a web storage to put your model files.
-
-Once you finish implementing a feature or bugfix, please send a Pull Request to https://github.com/espnet/espnet
+To make a report for `RESULTS.md`
+ - execute `show_result.sh` at a recipe main directory (where `run.sh` is located), as follows. 
+   You'll get environmental information and the evaluation result of each experiment in a markdown format.
+   ```
+   $ show_result.sh
+   ```
+ - execute `pack_model.sh` at the main directory of a recipe as follows. You'll get model information in a markdown format
+   ```
+   $ pack_model.sh --lm <language model> --dict <dict> <tr_conf> <dec_conf> <cmvn> <e2e>
+   ```
+ - `pack_model.sh` also produces a packed espnet model (`model.tar.gz`). If you upload this model to somewhere with a download link,
+   please put the link information to `RESULTS.md`.
+ - please contact Shinji Watanabe <shinjiw@ieee.org> if you want web storage to put your model files.
+
+#### 1.3.2 ESPnet2 recipes
+
+ESPnet2's recipes correspond to `egs2`. ESPnet2 applies a new paradigm without dependencies of Kaldi's binaries, which makes it lighter and more generalized.
+For ESPnet2, we do not recommend preparing recipe's stages for each corpus but rather using the common pipelines we provided in `asr.sh`, `tts.sh`, and 
+`enh.sh`. For details of creating ESPnet2 recipes, please refer to [egs2-readme](https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/README.md).
+
+The common pipeline of ESPnet2 recipes will take care of the `RESULTS.md` generation, model packing, and uploading. ESPnet2 models are maintained at Zenodo.
+To upload your model, you need first:
+1. Sign up to Zenodo: https://zenodo.org/
+2. Create access token: https://zenodo.org/account/settings/applications/tokens/new/
+3. Set your environment: % export ACCESS_TOKEN="<your token>"
+
+#### 1.3.3 Additional Checklist when building recipes
+
+- [ ] common files are linked with symlink: we ask to use symlink to refer common scripts and utilities (e.g., `utils`, `steps`, `asr.sh`). Please use `ln -sf <source_path> <target path>` rather than a new copy.
+- [ ] configuration files are formated: major configuration file is named as `conf/train.yaml` and `conf/decode.yaml` while other options are kept in `conf/tuning`
+- [ ] RESULT.md prepare: results are updated and prepared follow 1.3.1 or 1.3.2
+- [ ] corpus registration: please register your target corpus in https://github.com/espnet/espnet/blob/master/egs2/README.md or https://github.com/espnet/espnet/blob/master/egs/README.md and `db.sh` (for ESPnet2 only)
```

Comment:
- If a recipe for a new corpus is proposed, you should add its name and information to: https://github.com/espnet/espnet/blob/master/egs/README.md if it's a ESPnet1 recipe, or https://github.com/espnet/espnet/blob/master/egs2/README.md + `db.sh` if it's a ESPnet2 recipe.